google/cloud/bigquery/_pyarrow_helpers.py

```diff
@@ -68,6 +90,9 @@ def pyarrow_timestamp():
         "STRING": pyarrow.string,
         "TIME": pyarrow_time,
         "TIMESTAMP": pyarrow_timestamp,
+        "RANGE<TIMESTAMP>": pyarrow_range_timestamp,
```

Comment:
addmittedly I'm not super familiar with this code, but it seems like we're ending up declaring N scalar versions of range.  Do we get a simpler story from treating range more like a complex type and branching our struct processing logic?

Reply:
Indeed. I changed the code and opted to add a new conversion function, parallel to scalar and struct. I chose this instead of using the existing function for struct, because RANGE has fixed field names, so the logic is simpler.